src/handlers.cpp

```diff
@@ -623,6 +623,8 @@ void ImpiRegistrationStatusTask::run()
       _visited_network = _dest_realm;
     }
     _authorization_type = _req.param("auth-type");
+    std::string sos = _req.param("sos");
+    _emergency = sos == "true" ? true : false;
     TRC_DEBUG("Parsed HTTP request: private ID %s, public ID %s, visited network %s, authorization type %s",
```

Comment:
I'll leave this as is because the rest of the file deals with strings and not constants. I like your idea though.